src/BlocklyInterface/robotblocks/actuators/DcMotor.ts

```diff
@@ -43,19 +43,18 @@ export function addDcMotorBlock() {
       },
     ],
     (block) => {
-      var number_port = block.getFieldValue("port");
-      var dropdown_direction = block.getFieldValue("direction");
-      var value_power = JavaScript.valueToCode(
+      const numberPort = block.getFieldValue("port");
+      const dropdownDirection = block.getFieldValue("direction");
+      const valuePower = JavaScript.valueToCode(
         block,
         "power",
         JavaScript.ORDER_ATOMIC
       );
 
       // convert direction to power sign +/-
-      var isForward = dropdown_direction === "FORWARD";
-      let sign = isForward ? "1" : "-1";
-      var code = `setDcMotorPower(${number_port}, ${sign} * (${value_power}));\n`;
-      return code;
+      const isForward = dropdownDirection === "FORWARD";
+      const sign = isForward ? "1" : "-1";
```

Comment:
should sign not be a number instead of string

Reply:
It is a number, though it's going in a string - the output of the func is a string of a js function:
`"1" : "-1"` -> ```setDcMotorPower(1, 1 * (0.1));```

versus if it were attempting to output a string:
`"\"1\"" : "\"-1\""` -> ```setDcMotorPower(1, "1" * (0.1));```